designer/server/src/main/scala/pl/touk/nussknacker/ui/process/repository/ProcessActionRepository.scala

```diff
@@ -264,6 +264,22 @@ abstract class DbProcessActionRepository[F[_]](
     run(query.result.map(_.toSet))
   }
 
+  def getInProgressCanceledAndDeployed: F[Map[ProcessId, Set[ProcessActionType]]] = {
+    val query = processActionsTable
+      .filter(action =>
+        action.state === ProcessActionState.InProgress &&
+          action.actionType
+            .inSet(ProcessActionType.Deploy :: ProcessActionType.Cancel :: Nil)
+      )
+      .map(pa => (pa.processId, pa.actionType))
+    run(
+      query.result
+        .map(_.foldLeft(Map.empty[ProcessId, Set[ProcessActionType]]) { case (map, (processId, actionType)) =>
```

Comment:
Why not just do `toMap` on list of tuples?  We don't need to do this `getOrElse(Set.empty)` - it is already done in   `getProcessState`

Reply:
replaced with groupBy